### PR TITLE
[MRTK3] Added an option to use scalar distance instead of final displacement for bounds control toggle

### PR DESCRIPTION
**Pull request by akash14darshan**
## Overview
The toggling of bounds control is done when an asset is clicked but not moved.
The way we determine Clicked vs Moved is by using the asset's displacement. The initial position and final position.
Here is a video which demonstrates a case where the asset is moved significantly, yet determine the asset was clicked because we look only at the initial and final position, which are really close (lower than the drag threshold in this case)

https://user-images.githubusercontent.com/13174425/205303509-d4d12fc8-8a71-4e8f-b4c0-c30231c8d56f.mp4

## Changes
- Added code to calculate the scalar distance the object was moved (per frame). Using this logic, we can determine better if the asset was actually moved or not.
- Added a toggle to switch between the two modes. 

https://user-images.githubusercontent.com/13174425/205304552-c1b2f403-db30-4c7c-9722-130845a53149.mp4

PS: Do let me know if I need to change wordings/fieldname/comments anywhere since English is not my native language.

### CODE REVIEW

**Comment by akash14darshan** [comment]
@microsoft-github-policy-service agree

File: com.microsoft.mrtk.spatialmanipulation/BoundsControl/BoundsControl.cs [hunk 1]
```diff
@@ -544,14 +565,17 @@ private void OnDestroy()
 
         private void OnHostSelected(SelectEnterEventArgs args)
         {
+            isHostSelected = true;
```
Comment (keveleigh) [inline com.microsoft.mrtk.spatialmanipulation/BoundsControl/BoundsControl.cs:568]: Does this need to be set to `false` in `OnHostDeselected`?

Comment (akash14darshan) [reply com.microsoft.mrtk.spatialmanipulation/BoundsControl/BoundsControl.cs:568]: > Does this need to be set to `false` in `OnHostDeselected`?

@keveleigh yes my silly mistake. should be fixed now

**Review by RogPodge — APPROVED** [review]

**Comment by keveleigh** [comment]
@akash14darshan, sorry for such a delayed review here. This idea looks great! I think we'd like to change it slightly so that this is the default behavior (as well as updating the distance check to only run until the movement threshold is reached, since at that point there's no way to reverse). Do you mind if I push some commits to your branch? I'm also happy to make suggestions and let you run with it though!

**Comment by akash14darshan** [comment]
@keveleigh that's a good optimization and I totally agree with you.
Please feel free to make changes as you would like :D

**Comment by keveleigh** [comment]
Hey @akash14darshan, please take a look at the current state of this PR and make sure I haven't broken your original fix! I added some play mode tests to try to catch it, so it should be all good, but just want to make sure!

**Review by RogPodge — APPROVED** [review]

**Review by RogPodge — APPROVED** [review]

**Pull request merged**